process/block/baseProcess.go

```diff
@@ -433,6 +435,9 @@ func checkProcessorNilParameters(arguments ArgBaseProcessor) error {
 	if check.IfNil(arguments.CoreComponents.StatusHandler()) {
 		return process.ErrNilAppStatusHandler
 	}
+	if check.IfNil(arguments.GasHandler) {
```

Comment:
Maybe you should check also arguments.CoreComponents.EconomicsData() for nil

Reply:
done